lib/feature/one_hot_encoder.cc

```diff
@@ -23,26 +23,39 @@
 
 namespace clink {
 
-tfrt::AsyncValueRef<SparseVector> OneHotEncoderModel::transform(
-    const int value, const int column_index) const {
-  if (column_index >= model_data_.featuresizes_size()) {
-    return tfrt::MakeErrorAsyncValueRef("Column index out of range.");
-  }
+llvm::SmallVector<tfrt::RCReference<tfrt::AsyncValue>, 4>
+OneHotEncoderModel::transform(llvm::ArrayRef<tfrt::AsyncValue *> inputs,
+                              const ExecutionContext &exec_ctx) const {
+  auto output = MakeUnconstructedAsyncValueRef<SparseVector>(exec_ctx.host());
+  tfrt::EnqueueWork(exec_ctx, [model = this, async_value = inputs[0],
```

Comment:
We need to make sure that all variables (e.g. `model` and `async_value`) used by the lambda function will not be de-allocated before the lambda function ends. This is achieved for `output` by invoking `output.CopyRef()`. Should we do something similar for other variables?

Also, would it be simpler to do `value = async_value->get<int>()` in the lambda function's `[...]` directly? Same for other variables.

Reply:
I agree. I'll directly extract `value` from the async inputs and make lambda function have a counted reference to `model`.

One thing to notice is that along with this change, I also changed the `transform()` method to make it non-const. It is because that in `transform` we changed the reference count of the model itself.

Reply:
Sounds good.